R/layout.R

```diff
@@ -250,7 +250,75 @@ Layout <- ggproto("Layout", NULL,
     })
     names(label_grobs) <- names(labels)
     label_grobs
+  },
+
+  find_layer_mappings = function(self, plot) {
+    # Default mappings. Make sure it's a regular list instead of an uneval
+    # object.
+    default <- unclass(plot$mapping)
+
+    self$layer_mappings <- lapply(plot$layers, function(layer) {
+      mapping <- default
+      default[names(layer$mapping)] <- layer$mapping
```

Comment:
I can use `modifyList`, but since `layer$mapping` can be NULL, it also needs a `is.null` check.

Grepping for `modifyList` doesn't turn up anything that already does this.